### ISSUE TITLE
My connection is established, but the invoking function isn't working.

### ISSUE THREAD

**Issue by CaptainHunter** [adamhartford/SwiftR#89]
My connection is established, but the invoking function isn't working.

I have already tried invoking through a web browser and it's working fine, so I doubt the problem's from my SignalR hub. Would you kindly assist me again?

I am currently using signalR v2.2.1

<img width="294" alt="screen shot 2017-04-30 at 12 03 39 pm" src="https://cloud.githubusercontent.com/assets/28061999/25561377/3395c194-2d9d-11e7-9108-8ab0277280b9.png">

**Comment by LloydH28** [comment]
Good Day

I currently have the same issue has it ever been solved?

Regards

**Comment by adamhartford** [comment]
Could you share more code? Is it structured similar to the SwiftR demo app? I'm wondering if you have the hub declared such that it's no longer in scope by the time the `connect` callback is executed.

**Comment by LloydH28** [comment]
Hi thanks for your reply

I have structured it almost 100% like the demo except its all in a class that is created globally as signalR is used throughout the whole application so its done like this so that it runs once when the application has finished loading other values. 

What works is I can receive the events for on and signal r stays connected throughout the app i can recieve its posts but the second I Invoke I either get "Error: Object Reference not set to an instance of an object" and then have to click the button 2 or 3 times before the invoke actually goes through.

Do you know anything that could cause this. The signalR implimentation was setup by my companies previous developers with no documentation so I am unsure of what the issue could be. Like I said if I spam the send button it works now and again

Sorry for late reply our time-zones must be different

Regards
Lloyd Hendricks

**Comment by adamhartford** [comment]
Actually, `Object reference not set to an instance of an object` sounds like .NET runtime error, not Swift. If your transport is auto, and becomes WebSockets, remember that SwiftR doesn't support custom headers with WebSockets. It sounds possible that you're getting errors on the server side where headers are expected and not found.

**Comment by LloydH28** [comment]
Hi 

Thanks for the reply i will contact my API developer and ask him to confirm what the transport type is. What do you recommend using ?

Regards

**Comment by Maireno** [comment]
Hi.

I'm using your library for an iOS app. I have almost the same issue, but I made SignalR to use WebSockets with headers, but just the first time I established the connection with SignalR and I could use invoke to get info from the server without errors. In headers, I send an access token for my server.

After that, I restarted the app and it tried to establish a new connection. Then SignalR tried first ServerSentEvents, which didn't work with authorization.

Next, it tried LongPolling. The authorization was successful but every time I use Invoke I get the same error message: "Object reference not set to an instance of an object."

As you say that SwiftR doesn't support custom headers with WebSockets, how could it work the first time?

And after using WebSockets the first time, why is not using it the next time when I restart the app?

Could we say that LongPolling doesn't work on iOS?

Thank you in advance!

**Comment by zahid119** [comment]
Any solution so far? My connection to Hub is established and I have connectionId as well. But I'm unable to call any server method and get following error everytime. hubConnection
message = "There was an error invoking Hub method 'chathub.updateCustomer'.";

**Comment by munsifhayat** [comment]
@zahid119  did you find any solution for that. ?

**Comment by zahid119** [comment]
Yes. I accidentally set all my hub functions private. I updated them to public and it worked perfectly. The connection was always getting established because Connect and Disconnection functions were overridden from parent class.
